R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+#' Calculates a corrected life table from a mortAAR life table
+#'
+#' It is generally assumed that most skeletal populations lack the
+#' youngest age group. Life tables resulting from such populations
+#' will necessarily be misleading as they lead to believe that the
+#' mortality of younger children was lower than it actually was and
+#' that life expectancy was higher. For correcting these missing
+#' individuals, Bocquet-Appel and Masset conceived of several
+#' calculations based on regression analyses of modern comparable
+#' mortality data. However, the applicability of these indices
+#' to archaeological data is highly debated and does not necessarily
+#' lead to reliable results. Therefore, the correction needs to be
+#' weighted carefully and ideally only after the representativity of the
+#' base data has been checked with function lt.representativity.
+#'
+#' For the parameters see the documentation of \code{\link{life.table}}.
+#'
+#' @param life_table an object of class mortaar_life_table.
+#' @param agecor logical, optional.
+#' @param agecorfac numeric vector, optional.
+#' @param option_spline integer, optional.
+#'
+#' @return a list containing a data.frame with indices according to
+#' Bocquet-Appel and Masset showing the computed exact value as well
+#' as ranges and an object of class mortaar_life_table with the corrected
+#' values.
+#' \itemize{
+#'   \item \bold{e0}:   Corrected life expectancy.
+#'   \item \bold{1q0}:   Mortality of age group 0--1.
+#'   \item \bold{5q0}:   Mortality of age group 0--5.
+#'}
+#'
+#' @examples
+#' # Calculate representativity indices from real life dataset.
+#' library(magrittr)
+#' mag <- magdalenenberg
+#' mag <- mag %>% replace(mag == "60-x", "60-69")
+#' mag <- mag %>% tidyr::separate(a, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' mag_prep <- mag %>% prep.life.table(dec = "Dx",
+#' agebeg = "from", ageend = "to", method = "Equal5",
+#' agerange = "included")
+#' mag_result <- mag_prep %>% life.table()
+#' lt.correction(mag_result)
+#'
+#' @export
+#'
+lt.correction <- function(life_table, agecor = TRUE, agecorfac = c(), option_spline = NULL) {
+
+  indx <- lt.indices(life_table)
+
+  # corrected life expectancy at birth after Bocquet-Appel and Masset
+  e0 <- 78.721 * log10(sqrt(1 / indx$juvenile_i$d5_14_d20plus)) - 3.384
+  e0_range <- paste0(round(e0 - 1.503,1), "--", round(e0 + 1.503,1))
```

Comment:
Maybe it's better not to make the range a string value. When somebody wants to do anything else with this value beyond simply looking at it its's more convenient to be able to access it directly. 

Maybe something like this would be better:

|   | value | range_start | range_stop |
|---|---|---|---|
| e0   |   |   |   |
| 1q0 |   |   |   |
| 5q0  |   |   |   |

Reply:
Done!